Include/Rover/TrialView.hpp

```diff
@@ -0,0 +1,264 @@
+
+  //! Read-only random access reader for any Trial for a given Sample.
+  /*
+    \tparam S The type of the samples.
+  */
+  template<typename S>
+  class TrialView {
+    public:
+
+      //! The type of the samples stored in the Trial.
+      using Sample = S;
+
+    private:
+      using GetPtr = std::function<const Sample* (std::size_t)>;
+
+    public:
+
+      //! Random access constant iterator.
+      class ConstIterator {
+        public:
+          //! Increments the iterator.
+          ConstIterator& operator ++();
+
+          //! Increments the iterator.
+          ConstIterator operator ++(int);
+
+          //! Decrements the iterator.
+          ConstIterator& operator --();
+
+          //! Decrements the iterator.
+          ConstIterator operator --(int);
+
+          //! Advances the iterator.
+          ConstIterator& operator +=(std::ptrdiff_t offset);
+
+          //! Moves the iterator back.
+          ConstIterator& operator -=(std::ptrdiff_t offset);
+
+          //! Returns an iterator that is the result of advancing this.
+          ConstIterator operator +(std::ptrdiff_t offset) const;
+
+          //! Returns an iterator that is the result of moving this back.
+          ConstIterator operator -(std::ptrdiff_t offset) const;
+
+          //! Returns the offset between two iterators.
+          std::ptrdiff_t operator -(ConstIterator other) const;
+
+          //! Checks whether two iterators point to the same Sample.
+          bool operator ==(ConstIterator other) const;
+
+          //! Checks whether two iterators do not point to the same Sample.
+          bool operator !=(ConstIterator other) const;
+
+          //! Checks whether the Sample pointed to by the first iterator is to
+          //! the left of that pointed to by the second iterator.
+          bool operator <(ConstIterator other) const;
+
+          //! Checks whether the Sample pointed to by the first iterator is not
+          //! to the right of that pointed to by the second iterator.
+          bool operator <=(ConstIterator other) const;
+
+          //! Checks whether the Sample pointed to by the first iterator is to
+          //! the right of that pointed to by the second iterator.
+          bool operator >(ConstIterator other) const;
+
+          //! Checks whether the Sample pointed to by the first iterator is not
+          //! to the left of that pointed to by the second iterator.
+          bool operator >=(ConstIterator other) const;
+
+          //! Returns a reference to the Sample the iterator points to.
+          const Sample& operator *() const;
+
+          //! Returns a pointer to the Sample the iterator points to.
+          const Sample* operator ->() const;
+
+          //! Returns a reference to the Sample the iterator would point to
+          //! if it was advanced by the offset.
+          const Sample& operator [](std::ptrdiff_t offset) const;
+
+        private:
+          friend class TrialView<S>;
```

Comment:
The order of declarations are: friends, member variables, then member functions. Each section is separated by a blank line.